### PR TITLE
Fix invalid post warning when changing post types

### LINKED ISSUE #1655: Post button inappropriately mentions proposal options
How to replicate:
1) Start creating an Event type post
2) Change post type to Discussion
3) Add title and text but Post button stays disabled and tooltip talks about needing proposal options
4) Chanage post type to Offer and back to Discussion and Post button is enabled.

### PR DESCRIPTION
**Pull request by KevinTriplett**
Fixes #1655 

Hi Thomas -- again, I'm impressed with the work that has gone into proposals.

The root for this issue seems to be that isProposal is updated when the component renders but not before? It seems like isProposal is correct when the component renders, but not when `isValid` gets called.

(Moving the invalidPostWarning variable initialization to the PostEditor component was my first try but had to leave it there, not sure why. Putting it back into the ActionBar component class caused the message to not update at all.)

### CODE REVIEW

**Review by thomasgwatson — APPROVED** [review]
lgtm...

**Comment by KevinTriplett** [comment]
@thomasgwatson FYI discovered logic error and saw missing coverage for valid/invalid tests.

So I fixed the logic, protected against nulls, and wrote additional tests.

File: src/components/PostEditor/PostEditor.js [hunk 1]
```diff
@@ -447,19 +447,26 @@ class PostEditor extends React.Component {
       this.state.post,
       postUpdates
     )
-    const { isEvent, isProject, isProposal } = this.props
+
+    let validEvent = type?.length > 0
```
Comment (tibetsprague) [inline src/components/PostEditor/PostEditor.js:451]: shouldn't this be called validPost?

Comment (KevinTriplett) [reply src/components/PostEditor/PostEditor.js:451]: Ah, yikes ... This deals with post type data only (there are other tests for valid post), so I renamed it validTypeData.

**Review by tibetsprague — APPROVED** [review]

**Pull request merged**